src/main/java/org/minions/devfund/sergio/WhackAMole.java

```diff
@@ -0,0 +1,174 @@
+    private int score;
+    private int attemptsLeft;
+    private char[][] moleGrid;
+    private int gridDimension;
+
+    /**
+     * Constructor for WhackAMole.
+     *
+     * @param numAttempts   - The number of attempts for user.
+     * @param gridDimension - The dimension of the grid.
+     */
+    WhackAMole(int numAttempts, int gridDimension) {
+        attemptsLeft = numAttempts;
+        this.gridDimension = gridDimension;
+        moleGrid = new char[this.gridDimension][this.gridDimension];
+        loadGrid();
+    }
+
+    /**
+     * Return the current status of given position.
+     *
+     * @param posX - Represent the X coordinate.
+     * @param posY -Represent the Y coordinate.
+     * @return The current status.
+     */
+    char getStatusPosition(int posX, int posY) {
+        return moleGrid[posX][posY];
+    }
+
+    /**
+     * Gets the score.
+     *
+     * @return The score.
+     */
+    int getScore() {
+        return score;
+    }
+
+    /**
+     * Gets the number of current attempts.
+     *
+     * @return The number of attempts.
+     */
+    int getAttemptsLeft() {
+        return attemptsLeft;
+    }
+
+    /**
+     * Places a Mole.
+     *
+     * @param posX - Represent the X coordinate.
+     * @param posY - Represent the Y coordinate.
+     * @return True if the position (x,y) it is now a MOLE, false otherwise.
+     */
+    boolean place(int posX, int posY) {
+        if (moleGrid[posX][posY] != MOLE) {
+            moleGrid[posX][posY] = MOLE;
+            molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Whacks the grid attempting to hit a mole.
+     *
+     * @param posX - Represent the X coordinate.
+     * @param posY - Represent the Y coordinate.
+     */
+    void whack(int posX, int posY) {
+        char whack = 'W';
+
+        if (moleGrid[posX][posY] == MOLE) {
+            moleGrid[posX][posY] = whack;
+            molesLeft--;
+            score++;
+        }
+        attemptsLeft--;
+        printGridToUser();
+    }
+
+    /**
+     * Prints the current grid for user.
+     */
+    private void printGridToUser() {
+        System.out.println(getUserGrid());
+    }
+
+    /**
+     * Builds the user grid.
+     *
+     * @return A grid for user.
+     */
+    private StringBuilder getUserGrid() {
+        StringBuilder userGrid = new StringBuilder();
+        for (int i = 0; i < this.gridDimension; i++) {
+            for (int j = 0; j < this.gridDimension; j++) {
+                if (moleGrid[i][j] == 'M') {
+                    userGrid.append(GRID_WHITE_SPACE);
+                } else if (moleGrid[i][j] == 'W') {
+                    userGrid.append('W');
+                } else {
+                    userGrid.append(GRID_WHITE_SPACE);
+                }
+            }
+            userGrid.append("\n");
+        }
+        return userGrid;
+    }
+
+    /**
+     * Prints the grid.
+     */
+    public void printGrid() {
```

Comment:
Now this method is no longer necessary. it was only for edx.
you can delete it !